File: 백준/Bronze/1373. 2진수 8진수/2진수 8진수.cc
#include <iostream>
#include <string>

using namespace std;

int main(void) {
    string base_2;
    cin >> base_2;
    
    while (base_2.length() % 3 != 0) {
        base_2 = '0' + base_2;
    }
    
    for (int i = 0; i < base_2.length(); i += 3) {
        int digit = (base_2[i] - '0') * 4 + (base_2[i + 1] - '0') * 2 + (base_2[i + 2] - '0') * 1;
        cout << digit;
    }
    cout << '\n';
    return 0;
}

File: 백준/Bronze/1373. 2진수 8진수/2진수 8진수.py
base2 = input()

while (len(base2) % 3 != 0) :
    base2 = '0' + base2

base_8 = ""
for i in range(0, len(base2), 3) :
    digit = int(base2[i]) * 4 + int(base2[i + 1]) * 2 + int(base2[i + 2])
    base_8 = base_8 + str(digit)
    
print(base_8)